### PR TITLE
change simple socket server to simple server

### PR DESCRIPTION
**Pull request by charliesignalfx**

### CODE REVIEW

**Comment by charliesignalfx** [comment]
@keitwb fyi the integration tests failed on first run please see [this](https://circleci.com/gh/signalfx/signalfx-agent/5054)
They passed when I reran them.

**Comment by charliesignalfx** [comment]
@keitwb I may need your help debugging the doc issue.  I followed the instructions printed in the cirlce ci output.  I created the dev image, ran it, and ran "make docs" but it generates the output below.  

I've tried to debug this.  I also found where this error message is printed and added a word to the log message, but I can't get that word to appear.  I've tried remaking the dev image, but the word change doesn't show up.  Do I need to do something else to build the doc generator itself?

I appreciate any help you can offer here.

> bash -c "rm -f docs/{observers,monitors}/*"
scripts/docs/make-docs
[0001] ERROR Found MONITOR doc for monitor type collectd/activemq but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/apache but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/cassandra but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/consul but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/cpu but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/cpufreq but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/custom but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/df but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/disk but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/docker but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/elasticsearch but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/etcd but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/genericjmx but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/haproxy but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/health-checker but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/kafka but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/load but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/marathon but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/memcached but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/memory but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/signalfx-metadata but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/mongodb but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/mysql but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/interface but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/nginx but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/processes but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/protocols but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/rabbitmq but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/redis but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/spark but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/uptime but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/vmem but it doesn't appear to be registered
[0001] ERROR Found MONITOR doc for monitor type collectd/zookeeper but it doesn't appear to be registered
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x11803a4]
goroutine
 1 [running]:
github.com/signalfx/signalfx-agent/internal/selfdescribe.monitorDocsInPackage(0x0, 0x1a)
        /go/src/github.com/signalfx/signalfx-agent/internal/selfdescribe/monitors.go:90 +0x34
github.com/signalfx/signalfx-agent/internal/selfdescribe.monitorsStructMetadata.func1(0xc4200cf720, 0x1a, 0x1621f80, 0xc4204f2dd0, 0x0, 0x0, 0x0, 0xc4207836d0)
        /go/src/github.com/signalfx/signalfx-agent/internal/selfdescribe/monitors.go:48 +0xb9
path/filepath.walk(0xc4200cf720, 0x1a, 0x1621f80, 0xc4204f2dd0, 0xc4200f9060, 0x0, 0x0)
        /usr/local/go/src/path/filepath/path.go:361 +0xe5
path/filepath.walk(0x14e704a, 0x11, 0x1621f80, 0xc42000fd40, 0xc4200f9060, 0x0, 0x20)
        /usr/local/go/src/path/filepath/path.go:381 +0x2c2
path/filepath.Walk(0x14e704a, 0x11, 0xc4200f9060, 0x1, 0x1)
        /usr/local/go/src/path/filepath/path.go:403 +0x106
github.com/signalfx/signalfx-agent/internal/selfdescribe.monitorsStructMetadata(0x16337a0, 0x135f620, 0x11d16a3)
        /go/src/github.com/signalfx/signalfx-agent/internal/selfdescribe/monitors.go:43 +0xee
github.com/signalfx/signalfx-agent/internal/selfdescribe.JSON(0x1603ca0, 0xc4200bc010)
        /go/src/github.com/signalfx/signalfx-agent/internal/selfdescribe/selfdescribe.go:25 +0x23b
main.doSelfDescribe()
        /go/src/github.com/signalfx/signalfx-agent/cmd/agent/main.go:62 +0x42
main.main()
        /go/src/github.com/signalfx/signalfx-agent/cmd/agent/main.go:163 +0x235
Makefile:85: recipe for target 'docs' failed
make: *** [docs] Error 2

File: internal/core/config/config.go [hunk 1]
```diff
@@ -19,8 +19,8 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-// Config is the top level config struct that everything goes under
-type Config struct {
+// CommonConfig is the top level config struct for configurations that are common to all platoforms
+type CommonConfig struct {
```
Comment (keitwb) [inline internal/core/config/config.go]: This is backwards.  The main Config struct should stay the same and have the server fields embedded.  Or, even better, just use dynamic defaults provided by the [defaults library it is using](https://github.com/creasty/defaults) to set this at runtime based on whether it's running on Linux or Windows.

File: internal/core/diagnostics.go [hunk 2]
```diff
@@ -75,17 +75,17 @@ func (a *Agent) DiagnosticText() string {
...
-func (a *Agent) serveInternalMetrics(socketPath string) error {
-	if a.internalMetricsSocketStop != nil {
-		a.internalMetricsSocketStop()
+func (a *Agent) serveInternalMetrics(path string) error {
+	if a.internalMetricsServerStop != nil {
+		a.internalMetricsServerStop()
 	}
 
-	if err := os.MkdirAll(filepath.Dir(socketPath), 0700); err != nil {
+	if err := os.MkdirAll(filepath.Dir(path), 0700); err != nil {
```
Comment (keitwb) [inline internal/core/diagnostics.go]: What does this do on Windows with a pipe path?

Comment (charliesignalfx) [reply internal/core/diagnostics.go]: yeah actually i don't think I should do that on windows.  It explains some behavior I've been trying to understand through their doc on named pipes.  When I create the named pipe I see a directory called "pipe" created at the root of C:\.  I'll work around that on windows.

File: internal/utils/structtags/copy_to.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
+package structtags
+
+import (
+	"fmt"
+	"reflect"
+	"runtime"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
```
[the remark this answers is not in the document]
Comment (charliesignalfx) [reply internal/utils/structtags/copy_to.go]: will do

**Pull request merged**